src/NServiceBus.Transport.RabbitMQ.Tests/TestApprover.cs

```diff
@@ -16,18 +16,7 @@ public static void Verify(string text)
 
         class ApprovalNamer : UnitTestFrameworkNamer
         {
-            public ApprovalNamer()
-            {
-                var assemblyPath = GetType().Assembly.Location;
-                var assemblyDir = Path.GetDirectoryName(assemblyPath);
-
-                sourcePath = Path.Combine(assemblyDir, "..", "..", "..", "ApprovalFiles");
-            }
-
-            public override string SourcePath => sourcePath;
-
-            readonly string sourcePath;
+            public override string SourcePath { get; } = Path.Combine(TestContext.CurrentContext.TestDirectory, "..", "..", "..", "ApprovalFiles");
```

Comment:
I was going to drop the RHS down to the next line in my fixup, but I was sure you'd disagree. 😉